test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js

```diff
@@ -0,0 +1,388 @@
+  const transaction = await web3.eth.getTransaction(response.tx);
+  const gasUsed = new BN(response.receipt.gasUsed);
+  const gasPrice = new BN(transaction.gasPrice);
+  return gasUsed.mul(gasPrice);
+};
+
+/*
+ * Performs CoGateway.redeem.
+ *
+ * @return {Promise<void>}
+ */
+const redeem = async () => {
+  await approveCoGatewayForRedeemAmount(
+    ostPrime,
+    redeemRequest,
+    cogateway,
+  );
+
+  const initialBalancesBeforeRedeem = await redeemAssertion.captureBalances(
+    redeemRequest.redeemer,
+  );
+
+  const response = await cogateway.redeem(
+    redeemRequest.amount,
+    redeemRequest.beneficiary,
+    redeemRequest.gasPrice,
+    redeemRequest.gasLimit,
+    redeemRequest.nonce,
+    redeemRequest.hashLock,
+    {
+      from: redeemRequest.redeemer,
+      value: redeemRequest.bounty,
+    },
+  );
+
+  const transactionFeeInRedeem = await getTransactionFee(
+    response,
+    auxiliaryWeb3,
+  );
+
+  const event = EventDecoder.getEvents(response, cogateway);
+
+  await redeemAssertion.verify(
+    event,
+    redeemRequest,
+    transactionFeeInRedeem,
+    initialBalancesBeforeRedeem,
+  );
+  redeemRequest.messageHash = event.RedeemIntentDeclared._messageHash;
+};
+
+/*
+ * Performs Gateway.confirmRedeemIntent.
+ *
+ * @return {Promise<void>}
+ */
+const confirmRedeemIntent = async () => {
+  const { outboxProof, blockNumber } = await proveCoGateway();
+  redeemRequest.blockHeight = blockNumber;
+
+  const tx = await gateway.confirmRedeemIntent(
+    redeemRequest.redeemer,
+    redeemRequest.nonce,
+    redeemRequest.beneficiary,
+    redeemRequest.amount,
+    redeemRequest.gasPrice,
+    redeemRequest.gasLimit,
+    redeemRequest.blockHeight,
+    redeemRequest.hashLock,
+    outboxProof.storageProof[0].serializedProof,
+    { from: originAccounts[0] },
+  );
+
+  const event = EventDecoder.getEvents(tx, gateway);
+  ConfirmRedeemIntentAssertion.verify(event, redeemRequest);
+};
+
+/*
+ * Performs CoGateway.progressRedeemWithProof.
+ *
+ * @return {Promise<void>}
+ */
+const progressRedeemWithProof = async (messageStatus) => {
+  const initialBalanceBeforeProgress = await progressRedeemAssertion.captureBalances(
+    redeemRequest.redeemer,
+  );
+  const { inboxProof, blockNumber } = await proveGateway();
+  const response = await cogateway.progressRedeemWithProof(
+    redeemRequest.messageHash,
+    inboxProof.storageProof[0].serializedProof,
+    blockNumber,
+    messageStatus,
+    { from: redeemRequest.redeemer },
+  );
+
+  const transactionFeeInProgress = await getTransactionFee(
+    response,
+    auxiliaryWeb3,
+  );
+
+  const event = EventDecoder.getEvents(response, cogateway);
+  await progressRedeemAssertion.verify(
+    event,
+    redeemRequest,
+    transactionFeeInProgress,
+    initialBalanceBeforeProgress,
+    true,
+  );
+};
+
+/*
+ * Performs Gateway.progressUnstakeWithProof.
+ *
+ * @return {Promise<void>}
+ */
+const progressUnstakeWithProof = async (messageStatus) => {
```

Comment:
Similarly, update the documentation to include param `messageStatus `

Reply:
Done!